[Amos Chee Tian Ee] iP

amoschee opened the pull request:
# Hi there hello I'm Wiggly bot
## A bot that does stuff I imagine

** This bot was the first guy to **
- decide to decide things
- shoplift
- graduate
- put in a room


** This bot is able to **
1. pitch a movie
2. make producer guy happy 
3. make something super easy, barely an inconvenience

*Main code from producer guy*
``` Java
public static void main(String[] args) {
    System.out.println("So you have a chatbot for me? 😁");
}
```

** Bot TODO features **
- [x] able to do a backflip
- [x] able to snap bad guy's neck
- [x] able to save the day
- [ ] do better stuff
- [ ] sell more sandwiches with a pretty big pickle in it

> Is it going to be hard to think of a quote? - Producer guy
> Actually it's super easy, barely an inconvenience - ~~Screenwriter guy~~ Wiggly bot

[Link](https://www.youtube.com/watch?v=dQw4w9WgXcQ) to my repo: https://github.com/amoschee/ip

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

amoschee committed:
added level-0 increment

amoschee committed:
added Level-1 increment

amoschee committed:
added Level-2 increment

amoschee committed:
added Level-2 increment

amoschee committed:
added Level-2 increment

amoschee committed:
added Level-3 increment

amoschee committed:
added Level-4 increment

amoschee committed:
added A-TextUiTesting increment

amoschee committed:
added Level-5 increment

amoschee committed:
added Level-6 increment

amoschee committed:
Revert "added Level-6 increment"

This reverts commit d1dae047cb02f65c5dfb4b652b9f22505253b4c5.

amoschee committed:
Reapply "added Level-6 increment"

This reverts commit c41638331c5652e81a14fbc748ec9a6596426542.

amoschee committed:
added A-Enums increment

amoschee committed:
Add save support:
Save file located at data/Wiggly.txt

amoschee committed:
Empty commit to force a merge commit

amoschee committed:
Merge branch 'branch-Level-7'

amoschee committed:
Add Date support for deadline and events

amoschee committed:
Merge branch 'branch-Level-8'

amoschee committed:
Add more OOP:
Rename FileEditor.java to Storage.java
Separate UI display to Ui.java
Add Parser.java (partially implemented, not in use)

amoschee committed:
Add more OOP:
Rename FileEditor.java to Storage.java
Separate UI display to Ui.java
Add Parser.java (partially implemented, not in use)

amoschee committed:
Add more OOP:
Rename FileEditor.java to Storage.java
Separate UI display to Ui.java
Add Parser.java (partially implemented, not in use)

amoschee committed:
Add more OOP:
Rename FileEditor.java to Storage.java
Separate UI display to Ui.java
Add Parser.java
Separate Commands to different classes

amoschee committed:
Merge remote-tracking branch 'origin/branch-A-MoreOOP'

amoschee committed:
Divide classes into packages

amoschee committed:
Divide classes into packages

amoschee committed:
Merge branch 'branch-A-Packages'

amoschee committed:
Merge remote-tracking branch 'origin/add-gradle-support'

amoschee committed:
Add initial gradle support

amoschee committed:
Add Junit test cases

amoschee committed:
Merge branch 'branch-A-JUnit'

amoschee committed:
Package app as a Jar file

amoschee committed:
Merge remote-tracking branch 'origin/branch-A-Jar'

amoschee committed:
Add JavaDoc for task and utils package

amoschee committed:
Add JavaDoc for command package

amoschee committed:
Merge remote-tracking branch 'origin/branch-A-JavaDoc'

amoschee committed:
Change coding style to fit coding standard

amoschee committed:
Merge branch 'branch-A-CodingStandard'

amoschee committed:
Add find command

amoschee committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/wiggly/task/Task.java

karthickkc commented on src/main/java/wiggly/Wiggly.java:
```diff
@@ -0,0 +1,49 @@
+	}
+
+    public static void main(String[] args) {
+		new Wiggly("./data/Wiggly.txt").run();
```
Minor issue but indentation may be wrong here

karthickkc commented on src/main/java/wiggly/commands/AddCommand.java:
```diff
@@ -0,0 +1,75 @@
+	 * @param by The due date of the task, if empty, use {@code null} as the parameter
+	 * @param from The start date of the task, if empty, use {@code null} as the parameter
+	 * @param to The end date of the task, if empty, use {@code null} as the parameter
+	 */
```
You could add a blank line in the javadocs after description of method, before params for more readability

emmannyyy replied:
I second this 👍

karthickkc commented on src/main/java/wiggly/commands/AddCommand.java:
```diff
@@ -0,0 +1,75 @@
+
+		Task task;
+
+		switch (taskType) {
+		case TODO:
+			task = new ToDo(description);
+			ui.printWrappedString(taskList.addTask(task));
+			break;
+		case DEADLINE:
+			task = new Deadline(description, by);
+			ui.printWrappedString(taskList.addTask(task));
+			break;
+		case EVENT:
```
You may want to follow the style guide for switch case, as the standard for this course is slightly different from the usual way we would write switch case statements

karthickkc commented on src/test/java/wiggly/task/TaskTest.java:
Good job on testing extensively! 👍

karthickkc commented on src/main/java/wiggly/task/TaskType.java:
Nice use of ENUMS!

karthickkc reviewed (commented):
LGTM just some minor issues!

emmannyyy commented on src/main/java/wiggly/parser/Parser.java:
```diff
@@ -0,0 +1,79 @@
+public class Parser {
+
+
+	public static final Pattern USER_INPUT_PATTERN = Pattern.compile(
```
I like how clear this is! Great job on making this nice input pattern! Very neat, and the comments on the side are great for developers who may fork this code!

emmannyyy commented on src/main/java/wiggly/task/Event.java:
```diff
@@ -0,0 +1,36 @@
+        this.to = to;
+    }
+
+    @Override
```
Perhaps you may want to include some javaDoc comments to explain what this file format is, so that future developers can follow this format easily

emmannyyy commented on src/main/java/wiggly/util/Ui.java:
```diff
@@ -0,0 +1,61 @@
+	 * Prints out {@code END_LINE} first, followed by the specified string, and another {@code END_LINE}
+	 * @param str The string to be wrapped by {@code END_LINE}
+	 */
+	public void printWrappedString(String str) {
```
I love how clear your methods are! The naming conventions really follow the course's suggestions! Great job! 👍

emmannyyy reviewed (commented):
LGTM! I think you've done an amazing job! The naming conventions are really well done! Kudos!👍

amoschee committed:
Add style checker

amoschee committed:
Merge branch 'branch-A-CheckStyle'

amoschee committed:
Add Level-10 GUI

amoschee committed:
Merge branch 'branch-Level-10'

amoschee committed:
Add Level-10 GUI

amoschee committed:
Refactor code for improved readability

amoschee committed:
Merge pull request #1 from amoschee/branch-A-Assertions

amoschee committed:
Create CI workflows

amoschee committed:
Delete .github directory

amoschee committed:
Create main.yml

amoschee committed:
Merge pull request #2 from amoschee/branch-A-CodeQuality

Refactor code for improved readability

amoschee committed:
Add ArchiveCommand to archive tasks and enhance Storage with contentsToString method

This commit introduces a new ArchiveCommand class responsible for archiving the current tasks from Wiggly.txt into archive.txt within the same directory. This addition ensures that task data is preserved in an archived file format, providing better data management and recovery options.

Changes:

ArchiveCommand Class:

Added the ArchiveCommand class, which implements the functionality to archive tasks.
The execute method in ArchiveCommand performs the following:
Reads the content from Wiggly.txt using the contentsToString method in the Storage class.
Writes this content to archive.txt in the same directory.
Clears the current tasks from the task list to prepare for new entries.
Saves the cleared task list back to storage.
Added error handling to manage potential IOException during file operations, which is thrown as a WigglyException to maintain consistency in error reporting.
Storage Class Update:

Introduced a new method contentsToString in Storage.java.
The contentsToString method reads the entire content of the file associated with the Storage instance and returns it as a string. This method facilitates the ArchiveCommand class by providing a convenient way to retrieve file contents for archiving purposes.
Details:

ArchiveCommand Implementation:

Ensures that if there is no content to archive, a user-friendly message is returned indicating that there is nothing to archive.
Creates the archive.txt file if it does not already exist and writes the task content into it.
Properly manages file resources by closing the file writer after operations to prevent resource leaks.

Storage Class Enhancement:

The contentsToString method utilizes Files.readAllBytes to read the entire file content into a byte array and then converts it to a string. This approach simplifies the process of reading file content into the ArchiveCommand.

Impact: This commit improves task management by allowing for task data to be archived, thus preserving historical task information. The new method in the Storage class enhances modularity and code reuse by providing a straightforward mechanism to access file content as a string.

amoschee committed:
Merge pull request #4 from amoschee/branch-BCD-Extension

Add ArchiveCommand

amoschee committed:
Update README.md to include comprehensive user guide

amoschee committed:
Merge remote-tracking branch 'origin/branch-A-UserGuide'

amoschee committed:
Update README.md

amoschee committed:
Update README.md

amoschee committed:
Update README.md

amoschee committed:
Update README.md

amoschee committed:
Update README.md

amoschee committed:
Update README.md

amoschee committed:
Update README.md

amoschee committed:
Update README.md

amoschee committed:
Update README.md

amoschee committed:
Update README.md

amoschee committed:
Update README.md

amoschee committed:
Update README.md
